packages/amplify-graphql-schema-generator/src/input.ts

```diff
@@ -0,0 +1,88 @@
+import { print, InputObjectTypeDefinitionNode, DocumentNode } from 'graphql';
+import { ImportedRDSType } from '@aws-amplify/graphql-transformer-core';
+
+type AmplifyInputEntry = {
+  name: string;
+  type: string;
+  default: string | number;
+  comment?: string | undefined;
+};
+
+const getGlobalAmplifyInputEntries = (
+  transformerVersion: number,
+  dataSourceType = ImportedRDSType.MYSQL,
+  includeAuthRule = true,
+  authDocLink?: string,
+): AmplifyInputEntry[] => {
+  const inputs: AmplifyInputEntry[] = [
+    {
+      name: 'engine',
+      type: 'String',
+      default: dataSourceType,
+    },
+  ];
+
+  if (includeAuthRule && transformerVersion === 2) {
```

Comment:
Since this is the only place we're using this `transformerVersion` and we only support this flow for V2, we can get rid of this input if we throw an error early on.